holidays/countries/south_korea.py

```diff
@@ -87,66 +77,46 @@ def __init__(self, *args, **kwargs):
         ChristianHolidays.__init__(self)
         InternationalHolidays.__init__(self)
         StaticHolidays.__init__(self, cls=SouthKoreaStaticHolidays)
-        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_WORKDAY)
+        kwargs.setdefault("observed_since", 2014)
         super().__init__(*args, **kwargs)
 
-    def _add_alt_holiday(
-        self, dt: date, name: str = "", since: int = 2014, include_sat: bool = True
-    ) -> None:
-        """
-        Add alternative holiday on first day from the date provided
-        that's not already a another holiday nor a weekend.
-
-        :param dt:
-           The date of the holiday.
-
-        :param name:
-           The name of the holiday.
-
-        :param since:
-           Year starting from which alt holiday should be added
-
-        :param include_sat:
-           Whether Saturday is to be considered a weekend in addition to
-           Sunday.
-        """
-        if not self.observed:
-            return None
-
-        target_weekday = {SUN}
-        if include_sat:
-            target_weekday.add(SAT)
-        if (dt.weekday() in target_weekday or len(self.get_list(dt)) > 1) and dt.year >= since:
-            obs_date = dt + td(days=+1)
-            while obs_date.weekday() in target_weekday or obs_date in self:
-                obs_date += td(days=+1)
-            for name in (name,) if name else self.get_list(dt):
-                if "Alternative holiday" not in name:
+    def _populate_observed(self, dts: Set[date], spec_holidays: Dict[date, str]) -> None:
```

Comment:
Let's get rid of the ambiguity:
```suggestion
    def _populate_observed(self, dts: Set[date], three_day_holidays: Dict[date, str]) -> None:
```